src/Juvix/Compiler/Internal/Translation/FromInternal/Analysis/FunctionCall.hs

```diff
@@ -33,19 +37,25 @@ viewCall = \case
           lessThan = case viewExpressionAsPattern x of
             Nothing -> return (CallRow Nothing)
             Just x' -> do
-              s <- asks (findIndex (elem x') . (^. sizeSmaller))
+              s <- asks (findIndex (elem x') . (^. sizeSmaller) . findSizeInfo)
               return $ case s of
                 Nothing -> CallRow Nothing
                 Just s' -> CallRow (Just (s', RLe))
           equalTo :: Sem r CallRow
           equalTo =
             case viewExpressionAsPattern x of
               Just x' -> do
-                s <- asks (elemIndex x' . (^. sizeEqual))
+                s <- asks (elemIndex x' . (^. sizeEqual) . findSizeInfo)
                 return $ case s of
                   Nothing -> CallRow Nothing
                   Just s' -> CallRow (Just (s', REq))
               Nothing -> return (CallRow Nothing)
+          findSizeInfo :: SizeInfoMap -> SizeInfo
+          findSizeInfo infos =
+            fromMaybe (maybe emptySizeInfo snd . headMay $ infos ^. sizeInfoMap)
```

Comment:
I don't understand this line. Why is it ok to return the head even if the function name does not match?